src/operator/sum.rs

```diff
@@ -0,0 +1,201 @@
+//! N-ary plus operator.
+
+use crate::{
+    algebra::{AddAssignByRef, AddByRef, HasZero, WithNumEntries},
+    circuit::{
+        operator_traits::{NaryOperator, Operator},
+        Circuit, OwnershipPreference, Scope, Stream,
+    },
+};
+use std::{borrow::Cow, iter::once, mem::transmute, ops::Add};
+
+impl<P, D> Stream<Circuit<P>, D>
+where
+    P: Clone + 'static,
+    D: Add<Output = D> + AddByRef + AddAssignByRef + Clone + HasZero + WithNumEntries + 'static,
+{
+    /// Apply the [`Sum`] operator to `self` and all streams in `streams`.
+    pub fn sum<'a, I>(&'a self, streams: I) -> Stream<Circuit<P>, D>
+    where
+        I: Iterator<Item = &'a Self>,
```

Comment:
`IntoIterator` is more appropriate here
```suggestion
        I: IntoIterator<Item = &'a Self>,
```